core-war/src/main/java/org/silverpeas/web/pdc/control/PdcSearchSessionController.java

```diff
@@ -538,15 +544,19 @@ private void processFacetFiletype(Facet facet, GlobalSilverResult result) {
     }
   }
 
-  private void processFacetComponent(Facet facet, GlobalSilverResult result, List<String> blackList) {
+  private void processFacetComponent(Facet facet, GlobalSilverResult result, List<String> blackList,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3776)